### ISSUE TITLE
Tailscale doesn't update cleanly on Dietpi over ssh to the Tailscale IP

### ISSUE THREAD

**Issue by mstgrv** [tailscale/tailscale#9780]
I have a machine running Dietpi, and I've found that updating Tailscale doesn't complete cleanly when connected via ssh to the machine's tailscale IP. It only successfully updates over an ssh session to the physical LAN IP. 

When updating over the Tailscale IP, the update looks like it's completed correctly, but if you subsequently run 'apt update && apt upgrade', you get an error ('E: dpkg was interrupted, you must manually run 'sudo dpkg --configure -a' to correct the problem.'), but that command will only run successfully over ssh to the physical LAN IP. If just run the whole update over the physical LAN IP from the get go, the update does run cleanly and this error is never produced.

On another machine running Raspbian, there's no issue at all - updating Tailscale works fine over ssh to the Tailscale IP address, and this issue doesn't happen at all.

Any ideas what's happening with the Dietpi machine that would cause this issue? I posted more detail over at Dietpi's github page if that might help at all: https://github.com/MichaIng/DietPi/issues/6576

**Comment by MichaIng** [comment]
Okay let's keep investigation here were we possibly have some more insights/knowledge, if its okay for the Tailscale maintainers.

I just tested it and it works all well here:
- Installed Tailscale on a VM running locally here and a RISC-V VisionFive two running at a remote site.
- Connected both to my network via `tailscale up`.
- Connected via `ssh` from the RISC-V to the VM.
- Within the SSH session: `apt install tailscale=1.50.0` to downgrade, as well as afterwards `apt install tailscale` or `apt upgrade` to upgrade, the connection stays up 🤔. I see in the `postinst` script, that the service is restarted, but this does not affect the active connection.

It was from OpenSSH client to Dropbear server here, but since the service is not restarted, it doesn't matter anyway.

```console
root@VM-Bookworm:~# apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following packages will be upgraded:
  tailscale
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 25.0 MB of archives.
After this operation, 4096 B of additional disk space will be used.
Do you want to continue? [Y/n]
Get:1 https://pkgs.tailscale.com/stable/debian bookworm/main amd64 tailscale amd64 1.50.1 [25.0 MB]
Fetched 25.0 MB in 10s (2458 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 34123 files and directories currently installed.)
Preparing to unpack .../tailscale_1.50.1_amd64.deb ...
Unpacking tailscale (1.50.1) over (1.50.0) ...
Setting up tailscale (1.50.1) ...
2023-10-12 21:06:51 root@VM-Bookworm:~#
```

So the question is why the connection is aborted in case of @mstgrv. Probably there is some condition for this, which is not met in my case?

**Comment by kuraz758** [comment]
you mention that you used "tailscale up" and not "tailscale up --ssh" which might make the difference because with that option tailscale itself acts as an ssh server

**Comment by MichaIng** [comment]
Interesting fact. Since we have an SSH server pre-installed, it is not needed to have Tailscale acting as such, and, it would probably conflict on port 22 anyway? But it would be quite a good explanation if a service restart does not abort the Tailscale connection, but the embedded SSH server (and all its open connections).

Is there any logic in which case the embedded SSH server is automatically used, without `--ssh`? I was thinking whether it e.g. sees an OpenSSH server but e.g. does not know/recognise Dropbear. However, in my case, it worked all well with Dropbear as well. 

@mstgrv
Based on your report on our repo, you as well used `tailscale up` without `--ssh` flag, right?

**Comment by kuraz758** [comment]
the tailscale ssh server is used to simplify authentication and only binds to the tailscale ip

**Comment by MichaIng** [comment]
But never without explicitly adding the `--ssh` flag?

**Comment by kuraz758** [comment]
Yes, Tailscale only runs its own SSH server when you use the `--ssh` flag.
Without this flag, Tailscale's SSH functionalities are not enabled.

MichaIng ***@***.***> schrieb am Mo., 8. Jän. 2024, 19:12:
